src/main/java/com/coveros/selenified/element/check/State.java

```diff
@@ -0,0 +1,314 @@
+/*
+ * Copyright 2019 Coveros, Inc.
+ *
+ * This file is part of Selenified.
+ *
+ * Selenified is licensed under the Apache License, Version
+ * 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy
+ * of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.coveros.selenified.element.check;
+
+import com.coveros.selenified.OutputFile.Success;
+
+/**
+ * State extends Asserts to provide some additional verification capabilities.
+ * It will handle all verifications performed on the actual element. These
+ * asserts are custom to the framework, and in addition to providing easy object
+ * oriented capabilities, they take screenshots with each verification to
+ * provide additional traceability, and assist in troubleshooting and debugging
+ * failing tests. State checks that elements are in a particular state.
+ *
+ * @author Max Saperstone
+ * @version 3.1.0
+ * @lastupdate 9/13/2018
+ */
+public interface State extends Check {
+
+    // constants
+    String PRESENT = " is present on the page";
+    String NOTPRESENT = " is not present on the page";
+    String DISPLAYED = " is displayed on the page";
+    String NOTDISPLAYED = " is not displayed on the page";
+    String CHECKED = " is checked on the page";
+    String NOTCHECKED = " is not checked on the page";
+    String EDITABLE = " is editable on the page";
+    String NOTEDITABLE = " is not editable on the page";
+    String ENABLED = " is enabled on the page";
+    String NOTENABLED = " is not enabled on the page";
+
+    // ///////////////////////////////////////
+    // assessing functionality
+    // ///////////////////////////////////////
+
+    /**
+     * Verifies that the element is present. If the element isn't present, it
+     * waits up to the default time (5 seconds) for the element, before marking
+     * this verification as a failure. This information will be logged and
+     * recorded, with a screenshot for traceability and added debugging support.
+     */
+    void present();
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Rename method "present" to prevent any misunderstanding/clash with field "PRESENT" defined on line 40. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS1845)